File: Leetcode/119-Pascal's-Triangle-II/PascalTriangleII.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
    if (rowIndex<0) {
        return []
    }
    if (rowIndex==0) {
        return [1]
    }
    var cur = [1,1]
    if (rowIndex==1) {
        return cur
    }
    for (var k=2; k<=rowIndex; k++) {
        var next = []
        for (var i=0; i<=k; i++) {
            if (i==0 || i==k) {
                next.push(1)
                continue
            }
            next.push(cur[i-1] + cur[i])
        }
        cur = next
    }
    return cur
};

File: Leetcode/119-Pascal's-Triangle-II/PascalTriangleII.go
func getRow(rowIndex int) []int {
	if rowIndex < 0 {
		return make([]int, 0)
	}
	if rowIndex == 0 {
		return []int{1}
	}
	cur := make([]int, 2)
	cur[0], cur[1] = 1, 1
	if rowIndex == 1 {
		return cur
	}
	for k := 2; k <= rowIndex; k++ {
		next := make([]int, k+1)
		next[0], next[k] = 1, 1
		for i := 1; i < k; i++ {
			next[i] = cur[i-1] + cur[i]
		}
		cur = next
	}
	return cur
}
